compiler/ballerina-parser/src/main/java/io/ballerina/compiler/internal/parser/BallerinaParser.java

```diff
@@ -14165,7 +14165,15 @@ private STNode parseTypedBindingPatternOrMemberAccessRhs(STNode typeDescOrExpr,
             case CLOSE_BRACE_TOKEN: // T[a]}
             case COMMA_TOKEN:// T[a],
                 if (context == ParserRuleContext.AMBIGUOUS_STMT) {
-                    keyExpr = STNodeFactory.createNodeList(member);
+                    if (member == null) {
+                        STNode missingVarRef = STNodeFactory.createSimpleNameReferenceNode(
+                                SyntaxErrors.createMissingToken(SyntaxKind.IDENTIFIER_TOKEN));
+                        keyExpr = STNodeFactory.createNodeList(missingVarRef);
+                        closeBracket = SyntaxErrors.addDiagnostic(closeBracket,
+                                DiagnosticErrorCode.ERROR_MISSING_KEY_EXPR_IN_MEMBER_ACCESS_EXPR);
+                    } else {
+                        keyExpr = STNodeFactory.createNodeList(member);
+                    }
```

Comment:
Inside the same method, there seems to be multiple places where we call `STNodeFactory.createNodeList(member);` with member possibly being null. 

Shall we fix those as well? :)

Reply:
+1